src/compiler/program.ts

```diff
@@ -1213,14 +1201,20 @@ namespace ts {
                 return oldProgram.structureIsReused;
             }
 
-            modifiedFilePaths = modifiedSourceFiles.map(f => f.newFile.path);
+            const modifiedFiles = modifiedSourceFiles.map(f => f.oldFile);

```

Comment:
I think you can get away with creating duplicate array by using every function on modifiedSourceFiles to find if oldFile isn't in modified files

Reply:
I don't know if I correctly understand what your comment means:
If you mean I should prefer `!modifiedSourceFiles.some((f) => f.oldFile === oldFile)` in the loop below, I don't think it's worth changing. That would need to allocate a closure for every lookup and basically does the mapping inplace just to avoid allocating a new (probably very small) array.